grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -879,63 +902,83 @@ public <T extends Throwable> GrpcServiceBuilder addExceptionMapping(
      * Adds the specified exception mapping that maps a {@link Throwable} to a gRPC {@link Status}.
      * The mapping is used to handle a {@link Throwable} when it is raised.
      *
-     * <p>Note that this method and {@link #exceptionMapping(GrpcStatusFunction)} are mutually exclusive.
+     * <p>Note that this method and {@link #exceptionMapping(GrpcStatusFunction)}
+     * are mutually exclusive.
+     *
+     * @deprecated Use {@link #addExceptionHandler(Class, GrpcExceptionHandlerFunction)} instead.
      */
+    @Deprecated
     public GrpcServiceBuilder addExceptionMapping(Class<? extends Throwable> exceptionType,
                                                   GrpcStatusFunction statusFunction) {
         requireNonNull(exceptionType, "exceptionType");
         requireNonNull(statusFunction, "statusFunction");
+        addExceptionHandler(exceptionMappings, exceptionType, statusFunction::apply);
+        return this;
+    }
 
-        checkState(this.statusFunction == null,
-                   "addExceptionMapping() and exceptionMapping() are mutually exclusive.");
+    /**
+     * Adds the specified exception mapping that maps a {@link Throwable} to a gRPC {@link Status}.
+     * The mapping is used to handle a {@link Throwable} when it is raised.
+     *
+     * <p>Note that this method and {@link #exceptionHandler(GrpcExceptionHandlerFunction)}
+     * are mutually exclusive.
+     */
+    @UnstableApi
+    public GrpcServiceBuilder addExceptionHandler(Class<? extends Throwable> exceptionType,
+                                                  GrpcExceptionHandlerFunction exceptionHandler) {
```

Comment:
It is not your work but I'm not sure `addExceptionHandler` should be in `GrpcServiceBuilder`. `GrpcExceptionHandlerFunction` may provide builder methods for except types. What do you think of moving `addExceptionHandler` to `GrpcExceptionHandlerFunction`?
```java
GrpcExceptionHandlerFunction
  .builder()
  .on(IllegalArgumentException.class, (ctx, ige, metata) -> ...)
  .on(EntityNotFoundException.class, otherGrpcExceptionHandlerFunction)
  .build()
  .orElse(fallbackGrpcExceptionHandlerFunction)
```

Reply:
We are preparing for the next release. Let me handle this part myself. 🙇‍♂️

Reply:
It seems nice.
I agree with your proposal.